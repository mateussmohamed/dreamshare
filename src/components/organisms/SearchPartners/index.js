import React from 'react'
import styled from 'styled-components'

import { Container, Row, Column } from 'atoms'
import { Button } from 'molecules'

const Title = styled.h1`
  ${props => props.theme.typography.montserratStyle};
  font-size: 22px;
  font-weight: bold;
  text-align: center;
  color: ${props => props.theme.colors.greyishBrown};
`

const SubTitle = styled.h4`
  ${props => props.theme.typography.robotoStyle};
  font-size: 16px;
  text-align: center;
  color: ${props => props.theme.colors.brownishGrey};
`

const Input = styled.input`
  padding: 10px 15px;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.3);
  background-color: #ffffff;
  border: 1px solid #d4d4d4;
  outline: 0;
  ${props => props.theme.typography.robotoStyle};
  font-size: 14px;
  @media only screen and (min-width: 320px) {
    margin: 15px 0;
  }

  @media only screen and (min-width: 768px) {
    margin: 0 15px;
    min-width: 300px;
  }
`

const Form = styled.form`
  @media only screen and (min-width: 320px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  @media only screen and (min-width: 768px) {
    display: flex;
    flex-direction: row;
  }
`

const ContainerStyled = styled(Container)`
  margin: 40px auto;
  @media only screen and (min-width: 1024px) {
    margin: 100px auto;
  }
`

const FlexColumn = styled(Column)`
  @media only screen and (min-width: 320px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`

const SearchPartners = () => (
  <ContainerStyled>
    <Row flexDirection="column" mx={0}>
      <FlexColumn width={[1, 1, 1]}>
        <Title>Crate your holiday activity</Title>
        <SubTitle>Hi! What are your holiday interests?</SubTitle>
        <Form>
          <Input placeholder="Enter your insterests" />
          <Button secondary>Search partners</Button>
        </Form>
      </FlexColumn>
    </Row>
  </ContainerStyled>
)

export default SearchPartners
