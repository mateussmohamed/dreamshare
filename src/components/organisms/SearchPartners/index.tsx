import React from 'react'

import styled from 'styled-components'

import {
  Container,
  Row,
  Column,
  Form,
  Input,
  Button,
  SearchTitle,
  SearchDescription,
} from '../../atoms'

const RowStyled = styled(Row)`
  @media only screen and (min-width: 320px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`

export const SearchPartners = () => (
  <Container my={[40, 90]}>
    <RowStyled flexDirection="column" mx={0}>
      <Column>
        <SearchTitle>Crate your holiday activity</SearchTitle>
        <SearchDescription>
          Hi! What are your holiday interests?
        </SearchDescription>
        <Form>
          <Input placeholder="Enter your interests" />
          <Button secondary>Search partners</Button>
        </Form>
      </Column>
    </RowStyled>
  </Container>
)
