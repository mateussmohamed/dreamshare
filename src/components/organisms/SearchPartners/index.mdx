---
name: SearchPartners
menu: Organisms
---

import { Playground, PropsTable } from 'docz'
import SearchPartners from './index'

# SearchPartners
<PropsTable of={SearchPartners} />

<Playground>
  <SearchPartners />
</Playground>
