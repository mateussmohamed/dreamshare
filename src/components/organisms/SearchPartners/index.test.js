import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import SearchPartners from './index'

describe('[<SearchPartners />]', () => {
  const wrap = (props = {}) => shallow(<SearchPartners {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({})

    expect(wrapper).toMatchSnapshot()
  })
})
