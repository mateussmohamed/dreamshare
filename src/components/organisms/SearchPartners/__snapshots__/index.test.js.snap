// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<SearchPartners />] matches to snapshot 1`] = `
<Box
  css={
    Object {
      "maxWidth": "1024px",
    }
  }
  mx="auto"
  my={
    Array [
      40,
      90,
    ]
  }
>
  <Row
    flexDirection="column"
    mx={0}
  >
    <SearchPartners__FlexColumn
      width={
        Array [
          1,
          1,
          1,
        ]
      }
    >
      <SearchPartners__Title>
        Crate your holiday activity
      </SearchPartners__Title>
      <SearchPartners__SubTitle>
        Hi! What are your holiday interests?
      </SearchPartners__SubTitle>
      <SearchPartners__Form>
        <SearchPartners__Input
          placeholder="Enter your interests"
        />
        <Button
          secondary={true}
        >
          Search partners
        </Button>
      </SearchPartners__Form>
    </SearchPartners__FlexColumn>
  </Row>
</Box>
`;
